File: src/main/java/leetcode/arrays/easy/NRepeatedElementInSize2NArray/solution.py
from collections import Counter
from typing import List


class Solution:
    def repeated_n_times(self, nums: List[int]) -> int:
        counter = Counter(nums)
        for el in counter:
            if counter[el] > 1:
                return el


solution = Solution()
print(solution.repeated_n_times([5, 1, 5, 2, 5, 3, 5, 4]))

File: src/main/java/leetcode/arrays/easy/NRepeatedElementInSize2NArray/Solution.java
package leetcode.arrays.easy.NRepeatedElementInSize2NArray;


/*
You are given an integer array nums with the following properties:

    nums.length == 2 * n.
    nums contains n + 1 unique elements.
    Exactly one element of nums is repeated n times.

Return the element that is repeated n times.

Input: nums = [2,1,2,5,3,2]
Output: 2
*/
public class Solution {

    public int repeatedNTimes(int[] nums) {
        boolean[] tmp = new boolean[10000];
        for (int num : nums) {
            if (tmp[num]) {
                return num;
            }
            tmp[num] = true;
        }
        return 0;
    }
}
